.github/workflows/gcr-cleaner.yml

```
@@ -5,20 +5,29 @@ on:
   schedule:
     - cron: '0 0 */1 * *' # runs daily
   workflow_dispatch: # allows for manual invocation
+env:
+  GCP_WORKLOAD_IDENTITY_PROVIDER: 'projects/765091727073/locations/global/workloadIdentityPools/hato-atama-workload-identity/providers/github'
+  GCP_SERVICE_ACCOUNT: 'gcr-cleaner@hato-atama.iam.gserviceaccount.com'
 
 jobs:
   gcr-cleaner:
     runs-on: 'ubuntu-latest'
+    permissions:
+      id-token: write
+      contents: read
     steps:
-#      # configure based on your registry
-#      - uses: 'docker/login-action@v2'
-#        with:
-#          username: ${{ secrets.DOCKERHUB_USERNAME }}
-#          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - id: 'auth'
+        name: 'Authenticate to GCP'
+        uses: google-github-actions/auth@v1.0.0
+        with:
+          workload_identity_provider: ${{env.GCP_WORKLOAD_IDENTITY_PROVIDER}}
+          service_account: ${{env.GCP_SERVICE_ACCOUNT}}
```

```suggestion
          service_account: ${{env.GCP_SERVICE_ACCOUNT}}
          create_credentials_file: false
```

https://github.com/dev-hato/hato-atama/actions/runs/3504350385/jobs/5869981611#step:3:16

```
Warning: The "create_credentials_file" option is true, but the current GitHub workspace is empty. Did you forget to use "actions/checkout" before this step? If you do not intend to share authentication with future steps in this job, set "create_credentials_file" to false.
```

`create_credentials_file` はfalseにしておいた方が良さそう。